src/libraries/System.Memory.Data/tests/BinaryDataTests.cs

```diff
@@ -614,6 +614,32 @@ public void ToStringReturnEmptyStringWhenBinaryDataEmpty()
             Assert.Equal(string.Empty, BinaryData.Empty.ToString());
         }
 
+        [Theory]
+        [InlineData(0)]
+        [InlineData(4)]
+        [InlineData(7)]
+        public void LengthReturnsNumberOfBytes(int count)
+        {
+            var data = BinaryData.FromBytes(new byte[count]);
+            Assert.Equal(count, data.Length);
+        }
+
+        [Fact]
+        public void BinaryDataEmptyIsEmpty()
+        {
+            Assert.True(BinaryData.Empty.IsEmpty);
```

Comment:
I'd add tests that both `new BinaryData(new byte[] { }))` and `new BinaryData("").IsEmpty` are both IsEmpty and Length==0

and test Length and IsEmpty for all of byte[], string and ROM.

Reply:
Done in c507b04e485284626bdd6794b1a6223f8a915c9c